### DISCUSSION TITLE
Tooltip onOpenChange being fired with controlled state

### DISCUSSION THREAD

**Discussion by skarensmoll** [radix-ui/primitives#1834] [Help] [answered]
Hi there!,

I have a tooltip with a controlled state and while testing the onOpenChange, I realised the event is being fired even if the open state does not change. is there a reason for this?

https://codesandbox.io/s/spring-hill-3wljqi?file=/App.js:345-357

Thanks a lot

**Comment by benoitgrelard** [comment, marked as answer]
This is how controlled components work:
- `onOpenChange` is the component telling you that the value is changing
- `open` is you controlling the value

Right now, you are not updating the value in `onOpenChange` so it just logs, but never changes.

**Comment by ivanbanov** [comment]
@benoitgrelard although the props that you just mentioned are doing what you described, the issue is when the component didn't change the `open` value but either way it fires the `onOpenChange`. Correct me if Im wrong, but if the `open` didn't change the event should not be fired, right?

**Reply by benoitgrelard** [reply]
It should, it's the opposite in fact. It should only fire if it changes internally, not if it changes externally.
So here, you hover the trigger, the component is telling you that it should open, therefore `onOpenChange` triggers passing `true`.

**Reply by ivanbanov** [reply]
Got it, but then if the user hover the trigger in a controlled tooltip and for some reason the tooltip should not open, in the mouse leave the `onOpenChange` should not fire because the open didn't change (as suggested in the prop's name), does it make sense?

**Reply by benoitgrelard** [reply]
I'm not sure I follow, are you seeing this in the demo above? I only see a log when hovering, now when the mouse leaves the trigger.

**Reply by ivanbanov** [reply]
@benoitgrelard it's working as expected, it was a mistake in the local code
